### PR TITLE
Feature/arts 620 link person testbranch

### PR DESCRIPTION
**Pull request by ben0deinde**
## Description
Addition of Test automation scripts.

### Changes
- [ARTS-620] - Link Person to User with Auth
### Checklist:

- [ ] Branch name follows convention (feature/arts-#-short-name OR fix/arts-#-short-name)
- [ ] I have included a short-meaningful title, description and changes for this PR

[ARTS-620]: https://ndph-arts.atlassian.net/browse/ARTS-620

### CODE REVIEW

File: api-tests/specs/practitioner-service/linkperson.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+const utils = require('../../common/utils')
+const fetch = require("node-fetch");
+const { expect } = require('chai');
+const practitionerEndpointUri = '/api/practitioner'
+const linkpersonEndpointUri = '/api/practitioner/{personId}/link'
+
+
+describe('As a user with Update Person permission I want to link a Person to an existing User So that they have the Persons roles when they login to the system', function () {
+
+    it('Given a Person exists who is not linked to a User AND a User exists that is not linked to a Person When I submit an API request to link a Person who is NOT linked to a User to a User who is not linked to a Person in this trial Then the Person and User are linked and I receive a success acknowledgement ', async () => {
```
Comment (darrengsmithson) [inline api-tests/specs/practitioner-service/linkperson.js:19]: The Given, When, then seems a little confusing...would this be better..
Given 
Person A exists who is not linked to a User AND User B exists that is not linked to a Person 
When I submit an API request 
to link a Person A to User B 
Then 
Person A and User B are linked and I receive a success acknowledgement

File: api-tests/specs/practitioner-service/linkperson.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        let fetchResponse2 = await fetch(linkpersonEndpointUri2, {
+            headers: headers,
+            method: 'POST',
+            body: JSON.stringify(linkUserJSON),
+        })
+        expect(fetchResponse2.status).to.equal(HttpStatus.UNPROCESSABLE_ENTITY)
+
+    });
+
+    it('When I submit an API request with missing Practitioner ID to link a Person who is NOT linked to a User to a User who is not linked to a Person in this trial Then the Person and User are NOT linked and I receive a failure acknowledgement ', async () => {
```
Comment (darrengsmithson) [inline api-tests/specs/practitioner-service/linkperson.js:71]: would this be clearer?
Given a Person that is NOT linked to a User and a User who is not linked to a Person in this trial 

When I submit an API request with missing Practitioner ID to link the User and Person

Then the Person and User are NOT linked and I receive a failure acknowledgement

File: api-tests/specs/practitioner-service/linkperson.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        let fetchResponse3 = await fetch(linkpersonEndpointUri3, {
+            headers: headers,
+            method: 'POST',
+            body: JSON.stringify(linkUserJSON),
+        })
+        expect(fetchResponse3.status).to.equal(HttpStatus.UNPROCESSABLE_ENTITY)
+
+    });
+
+    it('When I submit an API request with missing User ID to link a Person who is NOT linked to a User to a User who is not linked to a Person in this trial Then the Person and User are NOT linked and I receive a failure acknowledgement ', async () => {
```
Comment (darrengsmithson) [inline api-tests/specs/practitioner-service/linkperson.js:98]: Given a Person who is NOT linked to a User to a User who is not linked to a Person in this trial 
When I submit an API request with missing User ID to link the Person and User 
Then the Person and User are NOT linked and I receive a failure
?

**Review by darrengsmithson — CHANGES_REQUESTED** [review]

File: api-tests/specs/practitioner-service/linkperson.js (hunk 2 above)
Comment (ben0deinde) [reply api-tests/specs/practitioner-service/linkperson.js:71]: Copied directly from the ARTS-185 story, the original link person story

File: api-tests/specs/practitioner-service/linkperson.js (hunk 3 above)
Comment (ben0deinde) [reply api-tests/specs/practitioner-service/linkperson.js:98]: Copied directly from the ARTS-185 story, the original link person story

**Review by darrengsmithson — APPROVED** [review]

**Review by sameerapurini — APPROVED** [review]

**Pull request merged**